tests/ui/missing_const_for_fn/cant_be_const.rs

```diff
@@ -68,3 +68,16 @@ impl std::ops::Add for Point {
         Point(self.0 + other.0, self.1 + other.1)
     }
 }
+
+pub struct A;
+impl A {
+    // This should not be const because the type implements `Drop`.
+    pub fn a(self) -> B {
+        B
+    }
```

Comment:
The same problem exists, if another argument of the function implements `Drop`. I think every argument needs to be checked.

```rust
impl B {
    pub fn b(self, c: C) -> B {
        B
    }
}

pub struct C;
impl Drop for C { fn drop(&mut self) {} }
```

Reply:
Oh right, that makes sense! I will have another look tomorrow :+1:

Reply:
Just as a note: this behavior is only true for arguments, that are moved into a function/method. References to types with `Drop` are perfectly okay (both for `self` and other parameters)